File: library/src/main/java/com/snail/commons/entity/CheckableItem.java
package com.snail.commons.entity;

import com.snail.commons.interfaces.Checkable;

/**
 * date: 2019/8/6 12:35
 * author: zengfansheng
 */
public class CheckableItem<T> implements Checkable<CheckableItem<T>> {
    private T data;
    private boolean isChecked;

    public CheckableItem() {
    }

    public CheckableItem(T data) {
        this.data = data;
    }

    public CheckableItem(T data, boolean isChecked) {
        this.data = data;
        this.isChecked = isChecked;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    @Override
    public boolean isChecked() {
        return isChecked;
    }

    @Override
    public CheckableItem<T> setChecked(boolean isChecked) {
        this.isChecked = isChecked;
        return this;
    }
}

File: library/src/main/java/com/snail/commons/entity/CheckableItem.kt
package com.snail.commons.entity

/**
 * 描述: 可选中
 * 时间: 2018/9/7 10:01
 * 作者: zengfansheng
 */
open class CheckableItem<T> @JvmOverloads constructor(var data: T? = null, var isChecked: Boolean = false)
